### ISSUE TITLE
Daily Rate Limit

### ISSUE THREAD

**Issue by DigitalFlow** [PokeAPI/pokeapi#263]
Hello everyone,

I have a question regarding the daily rate limit.
On the [documentation](http://pokeapi.co/docsv2/) it says that there is a daily rate limit of 300 requests per resource and IP address.
I'm writing an application that requires a "Pokedex" for working properly, i.E. I create a local XML with all Pokemon / Items / and so on once and save it, so that the API is only used once on initialization.
During initialization however, I often receive 429 - Too many requests, although I can't possibly hit the Daily Rate Limit, since I'm loading every ressource only once.
Are there other limits that I'm encountering here?
I load one page from the overview at a time, which has 20 record "links" on it.
I then load all 20 records in parallel / asynchronous, wait for all to finish and move on to the next page.
Could this trigger the 429 error and if yes, how is it supposed to be used then?

Kind Regards,
Florian

**Comment by jahed** [comment]
## Fair Use Policy

- Do not spam the entire service to scrape all the data at once - use the paginated pages.

**Comment by DigitalFlow** [comment]
Yes of course, I read that and as I understand that, my approach of retrieving the data page wise should be fine, since I'm using the paginated pages?

**Comment by jahed** [comment]
I'm not part of the PokeAPI team but I don't think that is what's meant by using paginated pages. If you're still just automatically scraping each URL, it's still scraping and kind of defeats the point of pagination.

If you're trying to rebuild the entire dataset into XML and bake it into your app for offline use, you can just use the CSV files in this repo and convert those.

**Comment by DigitalFlow** [comment]
I already reverted to ship the pregenerated xml with my tool, so there is no API usage as long as I don't trigger a regeneration of the files manually. I guess traffic wise this should be fine. 
A comment of the PokeApi regarding this topic would be nice though, as the only use of the overview pages in my opinion is to retrieve all resources.

**Comment by phalt** [comment]
Looks like you're hitting the rate limit. This is there to prevent scrapers. If you want the data that instantly then you're better off running a local instance of the API yourself. A better practice would be to request the data your need only when you need it, and store it when you retrieve it.

The paginated pages are there for navigating the content of the service :)
